Adds a note about superuser permission requiring staff permission as well. Fixes #2566 (#2567)
nautobot/nautobot: nautobot/docs/administration/permissions.md

| -                                                                      | Admin UI | API |
| ---------------------------------------------------------------------- | -------- | --- |

> | Superusers*                                                            | Yes      | Yes |
> | Staff users with `users.add_permission` or `users.change_permission`   | Yes      | Yes |
> | Regular users with `users.add_permission` or `users.change_permission` | No       | Yes |
> 
> Multiple permissions can be assigned to a user account.
> !!! info
> 
>     Superusers also required the Django Staff permission to view the Admin UI. 

Reviewer: Maybe:

```suggestion
| Staff superusers                                                         | Yes      | Yes |
| Non-staff superusers                                                     | No       | Yes |
| Staff users with `users.add_permission` or `users.change_permission`     | Yes      | Yes |
| Non-staff users with `users.add_permission` or `users.change_permission` | No       | Yes |

Multiple permissions can be assigned to a user account.
```
